### PR TITLE
[DATA-fix] Switch from using app address to signaling address

### PR DESCRIPTION
**Pull request by tahiyasalam**
**Context**
It seems like in the cloud config, the app address is empty whereas the signaling address is not. Instead, we should use the signaling address.

### CODE REVIEW

File: services/datamanager/datasync/sync.go [hunk 1]
```diff
@@ -56,7 +57,7 @@ type ManagerConstructor func(logger golog.Logger, cfg *config.Config, lastModMil
 
 // NewDefaultManager returns the default Manager that syncs data to app.viam.com.
 func NewDefaultManager(logger golog.Logger, cfg *config.Config, lastModMillis int) (Manager, error) {
-	if cfg.Cloud == nil || cfg.Cloud.AppAddress == "" {
+	if cfg.Cloud == nil || cfg.Cloud.SignalingAddress == "" {
```
Comment (AaronCasas) [inline services/datamanager/datasync/sync.go:60]: What is the signaling address?

Comment (tahiyasalam) [reply services/datamanager/datasync/sync.go:60]: The signaling address in the config is either "app.viam.com:port" or "app.viam.dev:port"

**Comment by agreenb** [comment]
@edaniels Checking on the difference between the signaling and app address? We're relying on the app address in other locations (e.g. the [package manager](https://github.com/viamrobotics/rdk/blob/d6cdd1c18fea9110e1ea08947b0ef4e5338bd397/robot/impl/local_robot.go#L464))

**Review by AaronCasas — APPROVED** [review]

File: services/datamanager/datasync/sync.go (hunk 1 above)
Comment (tahiyasalam) [reply services/datamanager/datasync/sync.go:60]: But maybe @edaniels can explain the difference whether signaling address and app address should be used

Comment (tahiyasalam) [reply services/datamanager/datasync/sync.go:60]: It seems like app address is often empty in the cloud config (ex: [here](https://app.viam.com/api/json1/config?id=4be27022-9d23-4fa5-8fa1-5d5fba16eefb&full=t&client=t) and [here](https://app.viam.com/api/json1/config?id=a6f8d7b1-8b39-4ac0-a52a-0808f970a80a&full=t&client=t))

Comment (edaniels) [reply services/datamanager/datasync/sync.go:60]: that seems like a bug

Comment (tahiyasalam) [reply services/datamanager/datasync/sync.go:60]: I went back through the robots I've created and noticed this is true

**Pull request closed without merging**